packages/navigator/src/browser/navigator-editor-sync.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2018 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject, postConstruct } from "inversify";
+import { FocusTracker, Widget } from "@phosphor/widgets";
+import { ApplicationShell } from '@theia/core/lib/browser/shell/application-shell';
+import { PreferenceChangeEvent, ISelectableTreeNode } from "@theia/core/lib/browser";
+import { FileNavigatorModel } from "./navigator-model";
+import { NavigatorConfiguration, NavigatorPreferences } from "./navigator-preferences";
+import { Navigatable } from "@theia/core/lib/browser/widgets/navigatable";
+
+@injectable()
+export class NavigatorEditorSynchronizer {
```

Comment:
Please merge it into `FileNavigatorContribution`?  It is the frontend application contribution and eagerly initialized.

Reply:
ok, I will merge them

Reply:
@akosyakov I stuck a bit with this. The problem is that I need then `FileNavigatorModel` in `FileNavigatorContribution`, but `FileNavigatorModel` is bound inside navigator child container.
I tried in `navigator-container` (but I am not sure that it is a good way to do it):
```
parent.bind(FileNavigatorModel).toDynamicValue(ctx => child.get(FileNavigatorModel));
```
but it didn't work (why?).
Or should I create another function to get  `FileNavigatorModel` from child container (then the child container will be singleton of course).
Or just return child container and do binding in parent one?
What is the right way here?

Reply:
Well, just rest could help. It is accessible via `widget.model`